app/scripts/metamask-controller.js

```diff
@@ -1053,6 +1054,48 @@ export default class MetamaskController extends EventEmitter {
       this.messageManager.clearUnapproved();
     });
 
+    this.metamaskMiddleware = createMetamaskMiddleware({
```

Comment:
This was copied directly from the network controller, with the parameters taken from the metamask-controller.js `initializeProvider` method (that's where they were hard-coded before).

I've decided to create this middleware _once_ and reference it in the RPC pipeline setup instead of constructing it per-connection because that's effectively what we were doing before. This does depend on these middleware all being stateless, but that was already an assumption we were relying on, at least partially.

It was reconstructed upon each network switch, whereas now it is not. But this _should_ be OK because most of this has nothing to do with the network.

There is one part that depends upon the network somewhat, and that's the `getPendingTransactionByHash` and `getPendingNonce` functions. But those each use the provider proxy provided by the network controller, so they should work the same as they do today between network switches.